docs/testplan/transceiver_onboarding/cmis_cdb_firmware_upgrade_testplan.md

```diff
@@ -0,0 +1,445 @@
+    <normalized_vendor_name_1>,<normalized_vendor_pn_1>,<firmware_version_2>,<firmware_binary_2>,<md5sum_2>
+    <normalized_vendor_name_1>,<normalized_vendor_pn_1>,<firmware_version_3>,<firmware_binary_3>,<md5sum_3>
+    # Add more vendor part numbers as needed
+    ```
+
+    For each firmware binary, the following metadata should be included:
+
+    - `normalized_vendor_name`: The normalized vendor name, created by applying the normalization rules described in the [CMIS CDB Firmware Binary Management](#111-cmis-cdb-firmware-binary-management) section.
+    - `normalized_vendor_pn`: The normalized vendor part number, created by applying the normalization rules described in the [CMIS CDB Firmware Binary Management](#111-cmis-cdb-firmware-binary-management) section.
+    - `fw_version`: The version of the firmware.
+    - `fw_binary_name`: The filename of the firmware binary.
+    - `md5sum`: The MD5 checksum of the firmware binary.
+
+4. A `cmis_cdb_firmware_base_url.csv` file (located in `ansible/files/transceiver_inventory` directory) should be present to define the base URL for downloading CMIS CDB firmware binaries. The file should follow this format:
+
+    ```csv
+    inv_name,fw_base_url
+    <inventory_file_name>,<base_url>
+    ```
+
+    - `inv_name`: The name of the inventory file that contains the definition of the target DUTs. For further details, please refer to the [Inventory File](https://github.com/sonic-net/sonic-mgmt/blob/master/docs/testbed/README.new.testbed.Configuration.md#inventory-file). The `inv_name` allows DUTs to be grouped based on their inventory file, enabling the test framework to fetch the correct base URL for firmware downloads.
+    - `fw_base_url`: The base URL from which the CMIS CDB firmware binaries can be downloaded. This URL should point to the directory where the firmware binaries are stored. e.g., `http://1.2.3.4/cmis_cdb_firmware/`.
+
+    Example of the file:
+
+    ```csv
+    inv_name,fw_base_url
+    lab,http://1.2.3.4/cmis_cdb_firmware/
+    ```
+
+5. A file (`sonic_{inv_name}_links.csv`) containing the connections of the ports should be present. This file is used to create the topology of the testbed which is required for minigraph generation.
+
+    - `inv_name` - inventory file name that contains the definition of the target DUTs. For further details, please refer to the [Inventory File](https://github.com/sonic-net/sonic-mgmt/blob/master/docs/testbed/README.new.testbed.Configuration.md#inventory-file)
+
+**Attributes for  the below tests**
+
+Following test attributes under `transceiver_firmware_upgrade` attribute category are applicable:
+
+| Attribute | Type | Default | Mandatory | Description |
+|-----------|------|---------|------------|-------------|
+| port_under_test | Dict | None | Yes | A dictionary containing the device name as the key and list of ports to be tested as its value |
```

Comment:
@az-pz Can you please refer to https://github.com/sonic-net/sonic-mgmt/pull/20230 and update this table accordingly since we may not need a dictionary for storing many attribute in this table.